### PR TITLE
Maintain common bookkeeping for component and stack analysis

### PR DESCRIPTION
**Pull request by samuzzal-choudhury**

### CODE REVIEW

File: bayesian/utils.py [hunk 1]
```diff
@@ -42,6 +54,10 @@ def server_create_analysis(ecosystem, package, version, api_flow=True, force=Fal
     :param force_graph_sync: force synchronization to graph
     :return: dispatcher ID handling flow
     """
+    # Bookkeeping first
+    component = MavenCoordinates.normalize_str(package) if ecosystem == 'maven' else package
+    server_create_component_bookkeeping(ecosystem, component, version, user_profile)
+
     args = {
         'ecosystem': ecosystem,
         'name': MavenCoordinates.normalize_str(package) if ecosystem == 'maven' else package,
```
Comment (fridex) [inline bayesian/utils.py]: This is already done above.

File: bayesian/utils.py [hunk 2]
```diff
@@ -31,8 +32,19 @@ def server_run_flow(flow_name, flow_args):
     Setup.connect_if_not_connected()
     return run_flow(flow_name, flow_args)
 
+def server_create_component_bookkeeping(ecosystem, name, version, user_profile):
+    args = {
+        'external_request_id': uuid.uuid4().hex,
+        'data': {
+            'api_name': 'component_analyses',
```
Comment (fridex) [inline bayesian/utils.py]: Will it be run only on `component_analyses`?

File: bayesian/auth.py [hunk 3]
```diff
@@ -15,6 +15,23 @@
 
 jwt.register_algorithm('RS256', RSAAlgorithm(RSAAlgorithm.SHA256))
 
+def decode_token():
+    token = request.headers.get('Authorization')
+    print (token)
```
Comment (samuzzal-choudhury) [inline bayesian/auth.py]: Removed. Changes are to be available in the next commit.

File: bayesian/auth.py [hunk 4]
```diff
@@ -15,6 +15,23 @@
 
 jwt.register_algorithm('RS256', RSAAlgorithm(RSAAlgorithm.SHA256))
 
+def decode_token():
```
Comment (samuzzal-choudhury) [inline bayesian/auth.py]: @fridex need more inputs perhaps. AFAIK, decode_token now includes the entire decoding token logic and the login_required wrapper does not duplicate any of that. As you can see, login_required uses decode_token.

The current design of the decode_token allows it to be used in other places like stack-analyses and component-analyses logic in api_v1.py.

File: bayesian/auth.py [hunk 5]
```diff
@@ -15,6 +15,23 @@
...
+    token = request.headers.get('Authorization')
+    print (token)
+    if token == None:
```
Comment (samuzzal-choudhury) [inline bayesian/auth.py]: Done. Changes are to be available in the next commit.

File: bayesian/api_v1.py [hunk 6]
```diff
@@ -251,24 +251,29 @@ class ComponentAnalyses(ResourceWithSchema):
...
+            
         if ecosystem == 'maven':
             package = MavenCoordinates.normalize_str(package)
         package = case_sensitivity_transform(ecosystem, package)
         result = get_analyses_from_graph(ecosystem, package, version)
-        current_app.logger.warn("%r" % result)
 
         if result is not None:
             # Known component for Bayesian
+            server_create_component_bookkeeping (ecosystem, package, version, decoded)
```
Comment (samuzzal-choudhury) [inline bayesian/api_v1.py]: Done. Changes are to be available in the next commit.

File: bayesian/api_v1.py [hunk 7]
```diff
@@ -472,6 +485,11 @@ class StackAnalysesV2(ResourceWithSchema):
 
     @staticmethod
     def post():
+        decoded = decode_token()
+        if decoded == None:
```
Comment (samuzzal-choudhury) [inline bayesian/api_v1.py]: @fridex good catch. Actually, this check is present in login_required. Need to check why that did not work.

File: bayesian/utils.py (hunk 2 above)
Comment (samuzzal-choudhury) [inline bayesian/utils.py]: `server_create_component_bookkeeping` function is used only for bookkeeping component-analyses.

File: bayesian/auth.py (hunk 4 above)
Comment (fridex) [inline bayesian/auth.py]: @samuzzal-choudhury OK, now I see. Thanks for clarification!

File: bayesian/utils.py (hunk 1 above)
Comment (samuzzal-choudhury) [inline bayesian/utils.py]: @fridex I am not repeating the call to server_create_component_bookkeeping. There are two scenarios for which server_create_component_bookkeeping is called separately.
1.  When component data is available on graphdb.
2. When component data is not available, componentApiFlow is called before calling bayesianFlow or bayesianApiFlow.

Did I miss anything here?

Comment (fridex) [inline bayesian/utils.py]: I meant the following line:

```python
MavenCoordinates.normalize_str(package) if ecosystem == 'maven' else package
```

In other words you can directly pass:

```python
args = {
  'name': component,
   ...
}

```

**Comment by samuzzal-choudhury** [comment]
Closing this PR in favour of the new PR https://github.com/fabric8-analytics/fabric8-analytics-server/pull/80

**Pull request closed without merging**